File: 프로그래머스/unrated/181873. 특정한 문자를 대문자로 바꾸기/특정한 문자를 대문자로 바꾸기.js
function solution(my_string, alp) {
    // 방법 2 - map, join 함수
    return [...my_string].map(el => {
        if (el == alp) return el.toUpperCase()
        else return el
    }).join('')
    
    // // 방법 1 - replaceAll 함수
    // return my_string.replaceAll(alp, alp.toUpperCase())
}

File: 프로그래머스/unrated/181873. 특정한 문자를 대문자로 바꾸기/특정한 문자를 대문자로 바꾸기.py
def solution(my_string, alp):
    # 방법 2 - 리스트 컴프리핸션
    return ''.join([ch.upper() if ch == alp else ch for ch in list(my_string)])
    
#     # 방법 1 - for문
#     answer = ''
    
#     for ch in my_string:
#         if ch == alp:
#             answer += ch.upper()
#         else:
#             answer += ch
            
#     return answer